Subsurface/Node/Common/Peripherals/inc/PWMInterface.h

```diff
@@ -0,0 +1,45 @@
+#include "stm32f3xx_hal.h"
+#include <cstdint>
+
+#ifndef PWMINTERFACE_H
+#define PWMINTERFACE_H
+
+class PWMInterface
+{
+public:
+    // Constructor
+    PWMInterface(uint64_t clockSpeed, uint32_t preScaler, uint32_t period,
+    uint32_t pulse, TIM_HandleTypeDef timer, uint32_t channel);
+
+    // Start and stop
+    void PWMStart();
+    void PWMStop();
+
+    // Set parameters
+    void setParameters(uint64_t clockspeed, uint32_t preScaler, uint32_t period,
+    uint32_t pulse, TIM_HandleTypeDef timer, uint32_t channel);
+
+    // Set Prescaler
+    void SET_PRESCALER();
+
+    // Set Autoload
+    void SET_AUTORELOAD();
+
+    // Set Compare
+    void SET_COMPARE();
```

Comment:
I would like to see these functions follow the same lower camel case convention as the rest ex: `void setPrescalar()`. Capitalized snake case is usually reserved for macros and the like